bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/internal/MqttBrokerConnection.java

```diff
@@ -128,6 +128,16 @@ public synchronized void start() throws Exception {
         }
 
         logger.info("Starting MQTT broker connection '{}'", name);
+
+        if (client != null) {
+            if ((!this.url.equals(client.getServerURI())) || (!this.clientId.equals(client.getClientId()))) {
+                logger.info("MQTT broker server URI or client ID changed. Removing previous connection..");
+                if (client.isConnected())
+                    client.disconnect();
```

Comment:
better put curly brackets around here. Shouldn't the logging be moved into this if as well?